manifests/port.pp

```diff
@@ -29,22 +30,11 @@
   Selinux::Port[$title] ->
   Anchor['selinux::end']
 
-  if $protocol {
-    validate_re($protocol, ['^tcp6?$', '^udp6?$'])
-    $protocol_switch = ['-p', $protocol]
-    $protocol_check = "${protocol} "
-    $port_exec_command = "add_${context}_${port}_${protocol}"
-  } else {
-    $protocol_switch = []
-    $protocol_check = '' # lint:ignore:empty_string_assignment variable is used to create regexp and undef is not possible
-    $port_exec_command = "add_${context}_${port}"
-  }
-
-  exec { $port_exec_command:
-    command => shellquote('semanage', 'port', $argument, '-t', $context, $protocol_switch, "${port}"), # lint:ignore:only_variable_string port can be number and we need to force it to be string for shellquote
-    # This works because there seems to be more than one space after protocol and before first port
-    unless  => sprintf('semanage port -l | grep -E %s', shellquote("^${context}  *${protocol_check}.* ${port}(\$|,)")),
-    path    => '/bin:/sbin:/usr/bin:/usr/sbin',
-    require => Class['selinux::package'],
+  validate_re($protocol, ['^tcp6?$', '^udp6?$'])
```

Comment:
is tcp6 or udp6 valid? `semanage port --help` on Fedora25 only shows udp, tcp, ipv4, ipv6.

But anyway the validate would be obsolete if using the data types.